frame/system/src/lib.rs

```diff
@@ -164,378 +92,136 @@ pub fn extrinsics_data_root<H: Hash>(xts: Vec<Vec<u8>>) -> H::Output {
+			+ Clone
+			+ OriginTrait<Call=Self::Call>;
+
+		/// The aggregated `Call` type.
+		type Call: Dispatchable + Debug;
+
+		/// Account index (aka nonce) type. This stores the number of previous transactions associated
+		/// with a sender account.
+		type Index:
+			Parameter + Member + MaybeSerialize + Debug + Default + MaybeDisplay + AtLeast32Bit
+			+ Copy;
+
+		/// The block number type used by the runtime.
+		type BlockNumber:
+			Parameter + Member + MaybeSerializeDeserialize + Debug + MaybeDisplay +
+			AtLeast32BitUnsigned + Default + Bounded + Copy + sp_std::hash::Hash +
+			sp_std::str::FromStr + MaybeMallocSizeOf;
+
+		/// The output of the `Hashing` function.
+		type Hash:
+			Parameter + Member + MaybeSerializeDeserialize + Debug + MaybeDisplay + SimpleBitOps + Ord
+			+ Default + Copy + CheckEqual + sp_std::hash::Hash + AsRef<[u8]> + AsMut<[u8]> + MaybeMallocSizeOf;
+
+		/// The hashing system (algorithm) being used in the runtime (e.g. Blake2).
+		type Hashing: Hash<Output=Self::Hash>;
+
+		/// The user account identifier type for the runtime.
+		type AccountId: Parameter + Member + MaybeSerializeDeserialize + Debug + MaybeDisplay + Ord
+			+ Default;
+
+		/// Converting trait to take a source type and convert to `AccountId`.
+		///
+		/// Used to define the type and conversion mechanism for referencing accounts in transactions.
+		/// It's perfectly reasonable for this to be an identity conversion (with the source type being
+		/// `AccountId`), but other modules (e.g. Indices module) may provide more functional/efficient
+		/// alternatives.
+		type Lookup: StaticLookup<Target=Self::AccountId>;
 
-		/// Hash of the previous block.
-		ParentHash get(fn parent_hash) build(|_| hash69()): T::Hash;
+		/// The block header.
+		type Header: Parameter + traits::Header<
+			Number=Self::BlockNumber,
+			Hash=Self::Hash,
+		>;
 
-		/// Digest of the current block, also part of the block header.
-		Digest get(fn digest): DigestOf<T>;
+		/// The aggregated event type of the runtime.
+		type Event: Parameter + Member + From<Event<Self>> + Debug + IsType<<Self as frame_system::Config>::Event>;
 
-		/// Events deposited for the current block.
-		Events get(fn events): Vec<EventRecord<T::Event, T::Hash>>;
+		/// Maximum number of block number to block hash mappings to keep (oldest pruned first).
+		#[pallet::constant]
+		type BlockHashCount: Get<Self::BlockNumber>;
 
-		/// The number of events in the `Events<T>` list.
-		EventCount get(fn event_count): EventIndex;
+		/// The weight of runtime database operations the runtime can invoke.
+		#[pallet::constant]
+		type DbWeight: Get<RuntimeDbWeight>;
 
-		// TODO: https://github.com/paritytech/substrate/issues/2553
-		// Possibly, we can improve it by using something like:
-		// `Option<(BlockNumber, Vec<EventIndex>)>`, however in this case we won't be able to use
-		// `EventTopics::append`.
+		/// Get the chain's current version.
+		type Version: Get<RuntimeVersion>;
```

Comment:
Wouldn't that be a good metadata candidate or is that exposed somewhere else, too?

Reply:
So that just means annotating with `#[pallet::constant]`?

Reply:
yup.

Reply:
> Wouldn't that be a good metadata candidate or is that exposed somewhere else, too?
It's available via rpc here:
https://github.com/paritytech/substrate/blob/9b08105b8c7106d723c4f470304ad9e2868569d9/client/rpc/src/state/mod.rs#L113

But I agree that it would also be useful as part of the metadata so have annotated it with `pallet::constant`